app/models/concerns/publishing_status.rb

```diff
@@ -0,0 +1,68 @@
+require "active_support/concern"
+
+module PublishingStatus
+  extend ActiveSupport::Concern
+
+  STATUSES = ["draft", "scheduled", "published"]
+
+  included do
+    validate :validate_publishing_status
+  end
+
+  def publishing_status
+    @publishing_status ||=
+      if published_at.nil?
+        "draft"
+      elsif published_at > Time.now
+        "scheduled"
+      else
+        "published"
+      end
+  end
+
+  def publishing_status_was
+    if published_at_was.nil?
+      "draft"
+    elsif published_at_was > Time.now
+      "scheduled"
+    else
+      "published"
+    end
+  end
+
+  def publishing_status=(value)
+    @publishing_status = value
```

Comment:
Adds getters/setters for the `publishing_status` virtual field.  Which allows the UX for the status-changer.